### PR TITLE
[move] Improvements for error propagation

### PR DESCRIPTION
**Pull request by georgemitenkov**
### Description

Attaches error messages to errors when constructing layouts or types.

### Test Plan

- [x] Added a test based on the bug report, inside APIs to ensure the message is propagated.

Replay verify: https://github.com/aptos-labs/aptos-core/actions/runs/8347861373

### CODE REVIEW

**Comment by georgemitenkov** [comment]
@runtian-zhou good point, I used the bug report and tested manually the error message is propagated when module publish fails on local network. Let me actually add this test here.

I don't think it is too hard to test, but achieving a good coverage can be tricky with the abundance test cases.

File: third_party/move/move-vm/runtime/src/interpreter.rs [hunk 1]
```diff
@@ -767,17 +775,39 @@ impl Interpreter {
...
+        current_frame: &Frame,
+    ) -> VMError {
+        // A verification error can be returned when
+        //   1) some check fails at runtime, e.g. type layout has too many type
+        //      nodes,
+        //   2) bytecode verifier fails, e.g. on module publishing.
+        // These errors mean that the code breaks some invariant, so we need to
+        // remap the error to VERIFICATION_ERROR which is of invariant violation
+        // type.
         if err.status_type() == StatusType::Verification {
```
Comment (brmataptos) [inline third_party/move/move-vm/runtime/src/interpreter.rs:791]: This seems kind of error-prone, as you're manually copying everything except the `StatusCode` from a `MVError` to a new one.  Why not a new function to `VMError` to replace the major status?  This outsourcing of functionality is probably why some is missing today.

Comment (georgemitenkov) [reply third_party/move/move-vm/runtime/src/interpreter.rs:790]: Good point, updated

**Review by davidiw — COMMENTED** [review]
@gregnazario any concerns about this breaking clients of the API? It seems to change the output of errors when reading transactions.

PR looks fine otherwise.

**Comment by georgemitenkov** [comment]
Thanks @davidiw for flagging that. I changed `Transaction Executed and Committed with Error` because it really is not committed, it fails in simulation, but would be great to know if this is an okay change

**Review by gregnazario — APPROVED** [review]
It seems fine to me, keeping in mind that this is a **breaking** change to wallets, or anything else that was parsing the message.

Let's ensure to have a proper communication plan around the change to ensure the experience doesn't break down.

**Pull request merged**